tests/tests.py

```diff
@@ -638,6 +638,26 @@ def test_365(self):
         categories = Category.objects.cache().annotate(newest_post=Subquery(newest_post[:1]))
         self.assertEqual(categories[0].newest_post, post.pk)
 
+    def test_385(self):
+        # case with settings.CACHEOPS_PICKLE_LIB == 'pickle' (by default)
+
+        Client.objects.create(name='Client Name')
+
+        with self.assertRaises(AttributeError) as e:
+            Client.objects.filter(name='Client Name').cache().first()
+        self.assertEqual(str(e.exception),
+                         "Can't pickle local object 'Client.__init__.<locals>.curry.<locals>._curried'")
+
+        invalidate_model(Client)
+
+        # case with settings.CACHEOPS_PICKLE_LIB == 'dill'
+        import dill
+        with mock.patch('pickle.loads', new=dill.loads), \
+             mock.patch('pickle.dumps', new=dill.dumps):
```

Comment:
If you implement settings more carefully you can use `override_settings()` here.